plutus-contract/test/Spec/MustSpendAtLeast.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE NumericUnderscores  #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeFamilies        #-}
+module Spec.MustSpendAtLeast(tests) where
+
+import Control.Monad (void)
+import Test.Tasty (TestTree, testGroup)
+
+import Ledger qualified
+import Ledger.Ada qualified as Ada
+import Ledger.Constraints.OffChain qualified as Constraints (MkTxError (OwnPubKeyMissing), plutusV1OtherScript,
+                                                             plutusV1TypedValidatorLookups, unspentOutputs)
+import Ledger.Constraints.OnChain.V1 qualified as Constraints (checkScriptContext)
+import Ledger.Constraints.TxConstraints qualified as Constraints (collectFromTheScript, mustIncludeDatum,
+                                                                  mustPayToTheScript, mustSpendAtLeast)
+import Ledger.Tx qualified as Tx
+import Ledger.Typed.Scripts qualified as Scripts
+import Plutus.Contract as Con
+import Plutus.Contract.Test (assertContractError, assertFailedTransaction, assertValidatedTransactionCount,
+                             checkPredicateOptions, defaultCheckOptions, w1, (.&&.))
+import Plutus.Trace qualified as Trace
+import Plutus.V1.Ledger.Api (Datum (Datum), ScriptContext, Validator, ValidatorHash)
+import Plutus.V1.Ledger.Scripts (ScriptError (EvaluationError))
+import PlutusTx qualified
+import PlutusTx.Prelude qualified as P
+import Prelude hiding (not)
+
+tests :: TestTree
+tests =
+    testGroup "MustSpendAtLeast"
+        [ entireScriptBalance
+        , lessThanScriptBalance
+        , higherThanScriptBalance
+        , phase2Failure
+        ]
+
+scriptBalance :: Integer
+scriptBalance = 25_000_000
+
+contract :: Integer -> Integer -> Contract () Empty ContractError ()
+contract offAmt onAmt = do
+    let lookups1 = Constraints.plutusV1TypedValidatorLookups typedValidator
+        tx1 = Constraints.mustPayToTheScript onAmt (Ada.lovelaceValueOf scriptBalance)
+    ledgerTx1 <- submitTxConstraintsWith lookups1 tx1
+    awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx1
+
+    utxos <- utxosAt scrAddress
+    let lookups2 = Constraints.plutusV1TypedValidatorLookups typedValidator
+            <> Constraints.plutusV1OtherScript validatorScript
```

Comment:
I don't think this is needed if you're using `Constraints.plutusV1TypedValidatorLookups typedValidator`